measure_extinction/modeldata.py

```diff
@@ -132,8 +153,44 @@ def __init__(
                         self.flux_uncs[cspec][k, i] = band_flux[1]
                 else:
                     # get the spectral data
-                    self.fluxes[cspec][k, :] = moddata.data[cspec].fluxes
-                    self.flux_uncs[cspec][k, :] = moddata.data[cspec].uncs
+                    if rebin is False:
+                        self.fluxes[cspec][k, :] = moddata.data[cspec].fluxes
+                        self.flux_uncs[cspec][k, :] = moddata.data[cspec].uncs
+                    else:
+                        # Initialize a temporary flux array
+                        full_fluxes = {}
+                        full_flux_uncs = {}
+                        full_fluxes[cspec] = np.zeros(
+                            (self.n_models, len(moddata.data[cspec].fluxes))
+                        )
+                        full_flux_uncs[cspec] = np.zeros(
+                            (self.n_models, len(moddata.data[cspec].fluxes))
+                        )
```

Comment:
This seems like it is in the wrong place.  Will this not re-initialize these variables for every value of `k`?  This does not seem like the correct behavior.

Reply:
Almost feel that the `self.n_models` can be removed and just do full_fluxes and full_flux_uncs for each model.  There is no need to have information about a previous model for rebinning, right?